src/components/organims/PrivacyBox.js

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import Auth0Lock from "auth0-lock";
+import { ApolloClient } from 'apollo-client';
+import { ApolloProvider } from 'react-apollo';
+import { InMemoryCache } from 'apollo-cache-inmemory';
+
+import PrivacyLogin from './privacy/PrivacyLogin';
+import Offboarding from './privacy/Offboarding';
+import OffboardingCancel from './privacy/OffboardingCancel';
+import OffboardingComplete from './privacy/OffboardingComplete';
+import OffboardingFail from './privacy/OffboardingFail';
+
+import { httpLink, authLink } from '../../utils/apollo';
+
+class PrivacyBox extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // Will hold Apollo Client
+    this.client = null;
+    // Init Auth0 Lock
+    this.lock = this.initLock();
+
+    // Bind necessary functions
+    this.onLogin = this.onLogin.bind(this);
+    this.onLogout = this.onLogout.bind(this);
+    this.navigate = this.navigate.bind(this);
+    this.handleAuthentication = this.handleAuthentication.bind(this);
+
+    // Init internal State
+    this.state = {
+      isAuthenticated: false,
+      profile: null,
+      route: null,
+    };
+  }
+  
+  initLock() {
+    return new Auth0Lock(process.env.AUTH0_CLIENT_ID, process.env.AUTH0_DOMAIN, {
+      autoclose: true,
+      auth: { redirect: false },
+      container: 'login-box',
+      theme: {
+        logo: 'https://assets.getstation.com/emails/facebook-profile.png',
+        primaryColor: '#164d7f',
+      },
+      languageDictionary: {
+        title: "Station"
+      },
+    });
+  }
+
+  async componentDidMount() {
+    // Closure to get rid of this binding problems
+    const lock = this.lock;
+    const handleAuthentication = this.handleAuthentication;
+
+    // Check user's session
+    lock.checkSession({}, function (error, authResult) {
+      if (error || !authResult) {
+        lock.show();
+      } else {
+        // User has an active session, so we can use the accessToken directly.
+        lock.getUserInfo(authResult.accessToken, handleAuthentication(authResult.idToken));
+      }
+    });
+
+    // Register authentication listener
+    lock.on("authenticated", authResult => {
+      lock.getUserInfo(authResult.accessToken, handleAuthentication(authResult.idToken));
+    });
+  }
+
+  handleAuthentication = (idToken) => (error, profile) => {
+    if (error) throw new Error(error.original);
+
+    // Init an Apollo Client
+    this.client = new ApolloClient({
```

Comment:
⚠️ you are mutating `this.client` and use in the `render`, you might end up having problems :)

Anyway, I think that you should instantiate a client. You could be able to do:
```js
const client = new  new ApolloClient({
  link: setContext((_, { headers }) => ({ authentication: this.state.authToken, ...headers}))
})
```